src/foam/mlang/MLang.java

```diff
@@ -194,10 +195,23 @@ public static Predicate HAS(Object o) {
       .build();
   }
 
-  public static Predicate HAS_PERMISSION(X userContext, String permissionPrefix) {
-    return new HasPermission.Builder(null)
+  // used by AuthenticatedDAO
+  public static Predicate HAS_PERMISSION(X userContext, Boolean remove, String permissionPrefix) {
```

Comment:
Do we still need this one?

Reply:
this will also be removed once migrated to authorizationdao

Reply:
ok